jetty-client/src/main/java/org/eclipse/jetty/client/util/ByteBufferContentProvider.java

```diff
@@ -57,6 +57,12 @@ public long getLength()
         return length;
     }
 
+    @Override
+    public boolean isRepeatable()
+    {
+        return true;
```

Comment:
I don't think that just because a content can technically be repeated, that it should be.

I think the user should pass in the constructor if it is repeatable and the default should be false.

Reply:
@gregw but the user may not know if a content needs to be repeatable. For example, with digest autentication the server may decide to "refresh" the tokens and send back a 401 on any random request. So either the client will always declare the content as repeatable, or if it does not it may fail.
At that point, the user has the burden of always specifying `repeatable=true`, while we can easily know that from the type of content and save the user some work.

Reply:
@sbordet then maybe `isRepeatable()` is the wrong name, perhaps it should be `isReproducable()`.

To me, `isRepeatable()` is kind of like `isIdempotent()`, meaning that the content is not something like a JSON encoded command to charge €1000 to my credit card.    

But in this case, we are not repeating the content, we are simply reproducing it because we know the first send of it was rejected with a 401.    If we were ever to write code that resent POSTs that failed silently, then we would need to check that the content was `isReproducable() && isRepeatable()`

We need detailed javadoc to say that this is not for resending a failed post, but only for known rejections.

Reply:
`isReproducible()` works for me as a name change, but I think that `ContentProvider` implementations that know that their content is reproducible should default to `true`, not forcing the user to pass this information all the times.

The javadocs will just specify that the method should return true if I can get another iterator out of the `ContentProvider` and get exactly the same bytes.
That is all.

How `isReproducible()` is used by the implementation may be explained, but I think it's out of scope for a `ContentProvider` javadoc. It's the implementation job to use it wisely.

Reply:
I agree that default to `true` is OK.    I think the documentation should just say that this field indicates that the content "can be" reproduced, not that it "should be" and that it is up to the caller to ensure safe reproduction of content.